otx/algorithms/visual_prompting/adapters/pytorch_lightning/datasets/dataset.py

```diff
@@ -0,0 +1,332 @@
+        """Generate bounding box from given mask.
+
+        Args:
+            gt_mask (np.ndarry): Mask to generate bounding box.
+            width (int): Width of image.
+            height (int): Height of image.
+
+        Returns:
+            List[int]: Generated bounding box from given mask.
+        """
+        y_indices, x_indices = np.where(gt_mask == 1)
+        x_min, x_max = np.min(x_indices), np.max(x_indices)
+        y_min, y_max = np.min(y_indices), np.max(y_indices)
+        return self.generate_bbox(x_min, y_min, x_max, y_max, width, height)
+
+
+class OTXVisualPromptingDataModule(LightningDataModule):
+    """Visual Prompting DataModule.
+
+    Args:
+        config (Union[DictConfig, ListConfig]): Configuration.
+        dataset (DatasetEntity): Dataset entity.
+    """
+
+    def __init__(self, config: Union[DictConfig, ListConfig], dataset: DatasetEntity) -> None:
+        super().__init__()
+        self.config = config
+        self.dataset = dataset
+
+        self.train_otx_dataset: DatasetEntity
+        self.val_otx_dataset: DatasetEntity
+        self.test_otx_dataset: DatasetEntity
+        self.predict_otx_dataset: DatasetEntity
+
+    def setup(self, stage: Optional[str] = None) -> None:
+        """Setup Visual Prompting Data Module.
+
+        Args:
+            stage (Optional[str], optional): train/val/test stages, defaults to None.
+        """
+        if not stage == "predict":
+            self.summary()
+
+        image_size = self.config.image_size
+        if isinstance(image_size, int):
+            image_size = [image_size]
+
+        if stage == "fit" or stage is None:
+            train_otx_dataset = self.dataset.get_subset(Subset.TRAINING)
+            val_otx_dataset = self.dataset.get_subset(Subset.VALIDATION)
+
+            # TODO (sungchul): distinguish between train and val config here
+            train_transform = val_transform = MultipleInputsCompose(
+                [
+                    ResizeLongestSide(target_length=max(image_size)),
+                    Pad(),
+                    transforms.Normalize(mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375]),
+                ]
+            )
+
+            self.train_dataset = OTXVIsualPromptingDataset(
+                train_otx_dataset, train_transform, offset_bbox=self.config.offset_bbox
+            )
+            self.val_dataset = OTXVIsualPromptingDataset(val_otx_dataset, val_transform)
+
+        if stage == "test":
+            test_otx_dataset = self.dataset.get_subset(Subset.TESTING)
+            test_transform = MultipleInputsCompose(
+                [
+                    ResizeLongestSide(target_length=max(image_size)),
+                    Pad(),
+                    transforms.Normalize(mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375]),
+                ]
+            )
+            self.test_dataset = OTXVIsualPromptingDataset(test_otx_dataset, test_transform)
+
+        if stage == "predict":
```

Comment:
Just for the reminder, this can be merged with state == "test"

Reply:
I checked it, but it seems impossible.
There are two reasons:
1. trainer.test is same with validate, so it outputs just metrics, not model outputs.
2. test callback doesn't receive model output, so we cannot process postprocessing for otx evaluation.